gbpservice/nfp/service_vendor_agents/haproxy/haproxy-agent/etc/dhcp/dhclient-exit-hooks.d/haproxy_routing

```diff
@@ -26,16 +25,28 @@ function ip2dec () {
 all_interfaces=`ifconfig | grep "Link encap" | grep -v "127.0.0.1" |cut -d: -d " " -f 1`;
 index=0;
 for i in $all_interfaces; do
-    if [ "$i" != "lo" ] #&& [ "$i" != "eth0" ]
-    #if [ "$i" == "$interface" ]
+    if [ "$i" != "lo" ] 
     then
-        ip_addr=`ip addr show $i | grep "inet " | grep -v "/32" | awk -F'[/ ]+' '{ print $3 }'`
+        interface_ip_configure_time=30
+        while [[ $interface_ip_configure_time -gt 0 ]]; do
+                ip_addr=`ip addr show $i | grep "inet " | grep -v "/32" | awk -F'[/ ]+' '{ print $3 }'`
+                if [[ -z $ip_addr ]]; then
+                        sleep 5
+                else
+                        break
+                fi
+        ((interface_ip_configure_time-=5))
+        done
+        if [[ -z $ip_addr ]]; then
+                echo "Hotplugged interface $i doesn't have an IP address. Hence, EXITING!!!" >&2
+                exit 1
+        fi
+
```

Comment:
Lets monitor the issue with this fix. If there is no issue, it is fine. Otherwise, we need to check if dhclient is running and run it again on the interface. When we tested manually, this step was done to get the default PBR back.